lib/auth/passwordReset.ts

```diff
@@ -0,0 +1,104 @@
+import { prisma } from "@lib/integration/prismaConnector";
+import { generateVerificationCode } from "./2fa";
+import { logMessage } from "@lib/logger";
+import { sanitizeEmailAddressForCognito } from "./cognito";
+import { getNotifyInstance } from "@lib/integration/notifyConnector";
+
+export class PasswordResetInvalidLink extends Error {}
+export class PasswordResetExpiredLink extends Error {}
+
+export const sendPasswordResetLink = async (email: string): Promise<void> => {
+  const sanitizedEmail = sanitizeEmailAddressForCognito(email);
+
+  const doesUserExist = await prisma.user.findUnique({
+    where: {
+      email: sanitizedEmail,
+    },
+  });
+
+  if (doesUserExist === null) throw new Error("UserDoesNotExist");
```

Comment:
Noticed that this results in throwing a 500 back to the client.  I'm wondering if we should just log that a password reset was requested with a wrong email address and do an early return?

Reply:
I could create a custom error to be treated differently on the API side. I did it for the other lib function because I knew we would need to handle it on the front end side (expired or invalid link).
Was not sure it would be necessary because the UI would not need to know exactly what happened. Maybe we want to show something special when the `UserDoesNotExist ` error happen though.

Reply:
Similar to our login we should try to keep information on whether or not a user exists internal as much as possible.  I wouldn't do any special handling... just a graceful fail and return a 200 like everything is normal.

Reply:
Sure thing! Will change it